src/constants/contracts/contracts.ts

```diff
@@ -1,8 +1,18 @@
-import { type ContractConfig } from '@wagmi/cli'
 import { type Abi, Address, erc20Abi, isAddress } from 'viem'
+import { mainnet, sepolia } from 'viem/chains'
 
 import { ENSRegistryABI } from '@/src/constants/contracts/abis/ENSRegistry'
-import { type ChainsIds, type RequiredChainId } from '@/src/lib/networks.config'
+import { type ChainsIds } from '@/src/lib/networks.config'
+
+type ValidateId<T> = T extends ChainsIds ? T : 'Invalid ID – This ID is not permitted'
```

Comment:
```suggestion
type ValidateId<T> = T extends ChainsIds ? T : never
```

Maybe with the string fails and you know the error, but. in that case should be possible to use `The Id ${T} is not permitted`